include/libtorrent/kademlia/node.hpp

```diff
@@ -75,22 +76,25 @@ class announce_observer : public observer
 	void reply(msg const&) { flags |= flag_done; }
 };
 
-struct udp_socket_interface
+struct socket_manager
 {
 	virtual bool has_quota() = 0;
-	virtual bool send_packet(entry& e, udp::endpoint const& addr) = 0;
+	virtual bool send_packet(aux::session_listen_socket* s, entry& e, udp::endpoint const& addr) = 0;
 protected:
-	~udp_socket_interface() {}
+	~socket_manager() {}
 };
 
+// get the closest node to the id with the given family_name
+using get_foreign_node_t = std::function<node*(node_id const&, std::string const&)>;
```

Comment:
Yes it can return a `nullptr` if there is no node with the requested address family.